lib/kubeclient/common.rb

```diff
@@ -87,11 +89,25 @@ def initialize_client(
       @http_max_redirects = http_max_redirects
       @as = as
 
+      @log = Logger.new(STDOUT)
+      @log.formatter = proc do |severity, datetime, progname, msg|
+          "#{datetime} [#{severity}]: #{msg}\n"
+      end
+
       if auth_options[:bearer_token]
         bearer_token(@auth_options[:bearer_token])
-      elsif auth_options[:bearer_token_file]
+      end
+      if auth_options[:bearer_token_file]
+        validate_bearer_token_file
+        @log.info("Reading bearer token from #{@auth_options[:bearer_token_file]}")
+        bearer_token(File.read(@auth_options[:bearer_token_file]))
+      elsif File.file?(DEFAULT_BEARER_TOKEN_FILE)
```

Comment:
I'd prefer to drop DEFAULT_BEARER_TOKEN_FILE from this PR.  This PR can support re-loading token but still require user to pass an explicit `:bearer_token_file`.

Why? Built-in support for standard in-cluster config is handy (it's silly that to this day it remains a recipe in README) but:

1. I'd prefer not to add functionality to v4.y branch that we don't have on master (unless you want to do both).

2. This only detects the token part, doesn't cover `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`.  I'm not sure if best way to support in-cluster config is auto-detection, vs. a separate factory e.g. `Kubeclient::Client.in_cluster()`.  If we don't need to decide these aspects here, we can land the re-loading part faster.

Reply:
Makes sense!
I'll remove DEFAULT_BEARER_TOKEN_FILE from this PR.